### DISCUSSION TITLE
Tree select/check value

### DISCUSSION THREAD

**Discussion by kuaniscreative** [Mezzanine-UI/mezzanine#67] [General]
~~Pardon me for poor English~~

在有子節點(非 leaf) 的情況時，select 或 check 的時候，最終的 value 需包含各個 node 的 value 還是只要 leaf 的 value？

舉例來說：分類 `drinks` 底下包含`soda`、`juice`、`tea`。當我點選 `drinks` 的 checkbox(這時候飲料、可樂、果汁、茶類都選取了)，最終的 value 會是 `['drinks', 'soda', 'juice', ''tea]` 還是 `['soda', 'juice', 'tea']`？

Ant Design 或 MMA 的做法是前者（包含節點 value）。還是說，要做一個可以 prop 讓這兩種行為可以被決定？

**Comment by fantasywind** [comment]
*leaf only*

Reasons:
Component user should know about structure between leafs and nodes, in most case, backend server only need smallest data id from mutation api.

**Reply by fantasywind** [reply]
But we can add optional prop to enable nodes' ids in value for specific cases.

**Comment by travor20814** [comment]
Agreed to @fantasywind suggestion.
Since we don't need to send node ids to backend server in most cases, I prefer "leaf only" for our implementation.

And, in my mind, the data structure will be something like:
```javascript
const value = ['soda', 'juice', 'tea'];

<Drink checked={exact ? value.includes('drink') : childrenAllSelected}>
  <Soda checked={value.includes('soda')} />
  <Juice checked={value.includes('juice')}  />
  <Tea checked={value.includes('tea')}  />
</Drink>
```
If we set `exact: true` on the node, it will be checked only when its node id appeared in source `value`, otherwise, we checked it base on its children selection. The structure will be like:

```javascript
const structure = [{
  key: 'drink',
  exact: true, // when set this to true, it will be checked only when 'drink' appeared in source `value`
  children: [{
    key: 'soda'
  }]
}];
```
or simply apply exact mode to all 
```javascript
<Tree exactMode />
```

Hmmmm... will this structure too complicated ?

**Comment by kuaniscreative** [comment]
Cool. I'll go with `exactMode` approach and default to not including node value.
